WordPress/src/main/java/org/wordpress/android/ui/stories/prefs/StoriesPrefs.kt

```diff
@@ -0,0 +1,128 @@
+            return true
+        }
+        try {
+            val inputStream = context.contentResolver.openInputStream(uri)
+            if (inputStream != null) {
+                inputStream.close()
+                return true
+            }
+        } catch (e: java.lang.Exception) {
+            e.printStackTrace()
+        }
+        return false
+    }
+
+    private fun saveSlide(context: Context, slideIdKey: String, storySlideJson: String) {
+        val editor = PreferenceManager.getDefaultSharedPreferences(context).edit()
+        editor.putString(slideIdKey, storySlideJson)
+        editor.apply()
+    }
+
+    @JvmStatic
+    fun isValidSlide(context: Context, siteId: Long, mediaId: RemoteMediaId): Boolean {
+        return checkSlideIdExists(context, siteId, mediaId) &&
+                checkSlideOriginalBackgroundMediaExists(context, siteId, mediaId)
+    }
+
+    private fun getSlideJson(context: Context, slideIdKey: String): String? {
+        return PreferenceManager.getDefaultSharedPreferences(context).getString(slideIdKey, null)
+    }
+
+    @JvmStatic
+    fun getSlideWithRemoteId(context: Context, siteId: Long, mediaId: RemoteMediaId): StoryFrameItem? {
+        val jsonSlide = getSlideJson(context, buildSlideKey(siteId, mediaId))
+        jsonSlide?.let {
+            return StorySerializerUtils.deserializeStoryFrameItem(jsonSlide)
+        } ?: return null
+    }
+
+    @JvmStatic
+    fun getSlideWithLocalId(context: Context, siteId: Long, mediaId: LocalMediaId): StoryFrameItem? {
+        val jsonSlide = getSlideJson(context, buildSlideKey(siteId, mediaId))
+        jsonSlide?.let {
+            return StorySerializerUtils.deserializeStoryFrameItem(jsonSlide)
+        } ?: return null
+    }
+
+    fun saveSlideWithLocalId(context: Context, siteId: Long, mediaId: LocalMediaId, storyFrameItem: StoryFrameItem) {
+        val slideIdKey = buildSlideKey(siteId, mediaId)
+        saveSlide(context, slideIdKey, StorySerializerUtils.serializeStoryFrameItem(storyFrameItem))
+    }
+
+    fun saveSlideWithRemoteId(context: Context, siteId: Long, mediaId: RemoteMediaId, storyFrameItem: StoryFrameItem) {
+        val slideIdKey = buildSlideKey(siteId, mediaId)
+        saveSlide(context, slideIdKey, StorySerializerUtils.serializeStoryFrameItem(storyFrameItem))
+    }
+
+    fun deleteSlideWithLocalId(context: Context, siteId: Long, mediaId: LocalMediaId) {
+        val slideIdKey = buildSlideKey(siteId, mediaId)
+        val editor = PreferenceManager.getDefaultSharedPreferences(context).edit()
+        editor.remove(slideIdKey)
+        editor.apply()
+    }
+
+    fun deleteSlideWithRemoteId(context: Context, siteId: Long, mediaId: RemoteMediaId) {
+        val slideIdKey = buildSlideKey(siteId, mediaId)
+        val editor = PreferenceManager.getDefaultSharedPreferences(context).edit()
+        editor.remove(slideIdKey)
+        editor.apply()
```

Comment:
We can simplify these with a bit of kotlin goodness :) 
```suggestion
        PreferenceManager.getDefaultSharedPreferences(context).edit().apply { 
            remove(slideIdKey)
            apply()
        }
```